code/script.js

```diff
@@ -1,47 +1,228 @@
 // All the DOM selectors stored as short variables
-const chat = document.getElementById('chat')
+const chat = document.getElementById("chat");
+const nameInput = document.getElementById("name-input");
+const submitBtn = document.getElementById("btn");
+const inputWrapper = document.getElementById("input-wrapper");
 
+//[updated] 'name' is reserved as global variable? instead 'name' -> 'username'
+let username;
 // Global variables, if you need any, declared here
 
 // Functions declared here
 
 // This function will add a chat bubble in the correct place based on who the sender is
 const showMessage = (message, sender) => {
-  if (sender === 'user') {
+  if (sender === "user") {
     chat.innerHTML += `
       <section class="user-msg">
         <div class="bubble user-bubble">
           <p>${message}</p>
         </div>
         <img src="assets/user.png" alt="User" />  
       </section>
-    `
-  } else if (sender === 'bot') {
+    `;
+  } else if (sender === "bot") {
     chat.innerHTML += `
       <section class="bot-msg">
         <img src="assets/bot.png" alt="Bot" />
         <div class="bubble bot-bubble">
           <p>${message}</p>
         </div>
       </section>
-    `
+    `;
   }
   // This little thing makes the chat scroll to the last message when there are too many to be shown in the chat box
-  chat.scrollTop = chat.scrollHeight
-}
+  chat.scrollTop = chat.scrollHeight;
+};
+
+// Restart function
+const restartPage = (botMessage, buttonMessage) => {
+  inputWrapper.innerHTML = `
+  <button id="restart">${buttonMessage}</button>
+   `;
+  showMessage(botMessage, "bot");
```

Comment:
Really liked this function! Nice thought to restart the chatbot and smart to do it in this way so you could change the messages depending on different parameters.